include/deal.II/matrix_free/face_setup_internal.h

```diff
@@ -0,0 +1,944 @@
+                        AssertIndexRange(f, faces_type[type].size()-vectorization_width+1);
+                        for (unsigned int v=0; v<vectorization_width; ++v)
+                          {
+                            macro_face.cells_interior[v] = faces_in[faces_type[type][f+v]].cells_interior[0];
+                            macro_face.cells_exterior[v] = faces_in[faces_type[type][f+v]].cells_exterior[0];
+                            touched[f+v] = 1;
+                          }
+                        new_faces.insert(macro_face);
+                        f += vectorization_width-1;
+                        n_vectorized += vectorization_width;
+                      }
+                  }
+
+              std::vector<unsigned int> untouched;
+              untouched.reserve(no_faces - n_vectorized);
+              for (unsigned int f=0; f<no_faces; ++f)
+                if (touched[f] == 0)
+                  untouched.push_back(f);
+              unsigned int v=0;
+              for (auto f : untouched)
+                {
+                  macro_face.cells_interior[v] = faces_in[faces_type[type][f]].cells_interior[0];
+                  macro_face.cells_exterior[v] = faces_in[faces_type[type][f]].cells_exterior[0];
+                  ++v;
+                  if (v == vectorization_width)
+                    {
+                      new_faces.insert(macro_face);
+                      v = 0;
+                    }
+                }
+              if (v > 0 && v < vectorization_width)
+                {
+                  // must add non-filled face
+                  if (hard_vectorization_boundary[partition+1] ||
+                      partition == face_partition_data.size()-2)
+                    {
+                      for ( ; v < vectorization_width; ++v)
+                        {
+                          // Dummy cell, not used
+                          macro_face.cells_interior[v] = numbers::invalid_unsigned_int;
+                          macro_face.cells_exterior[v] = numbers::invalid_unsigned_int;
+                        }
+                      new_faces.insert(macro_face);
+                    }
+                  else
+                    {
+                      // postpone to the next partition
+                      std::vector<unsigned int> untreated(v);
+                      for (unsigned int f=0; f<v; ++f)
+                        untreated[f] = faces_type[type][*(untouched.end()-1-f)];
+                      new_faces_type.push_back(untreated);
+                    }
+                }
+            }
+
+          // insert sorted list to vector of faces
+          for (auto it = new_faces.begin(); it != new_faces.end(); ++it)
+            faces_out.push_back(*it);
+          face_partition_data[partition+1] += new_faces.size();
+
+          // set the faces that were left over to faces_type for the next round
+          faces_type = std::move(new_faces_type);
+        }
+
+      // final safety checks
```

Comment:
Maybe move to `DEBUG`. I am surprise that the compiler does not complain about `nfaces` in release mode.